src/functions/routers/types.ts

```diff
@@ -0,0 +1,60 @@
+import {
+  FunctionContext,
+  FunctionDefinitionArgs,
+  FunctionParameters,
+  FunctionRuntimeParameters,
+} from "../types.ts";
+
+export type BlockActionHandler<Definition> = Definition extends
+  FunctionDefinitionArgs<infer I, infer O, infer RI, infer RO> ? {
+    (
+      context: ActionContext<FunctionRuntimeParameters<I, RI>>,
+    ): Promise<void> | void;
+  }
+  : never;
+
+export type ActionContext<InputParameters> =
+  & Omit<FunctionContext<InputParameters>, "event">
+  & ActionSpecificContext<InputParameters>;
+
+type ActionSpecificContext<InputParameters extends FunctionParameters> = {
+  body: BlockActionInvocationBody<InputParameters>;
+  action: BlockAction;
+};
+
+// TODO: all below stolen from deno-slack-runtime, need to centralize these somewhere
+export type BlockActionInvocationBody<
+  InputParameters extends FunctionParameters,
+> = {
+  type: string;
+  actions: BlockAction[];
+  bot_access_token?: string;
```

Comment:
We're exposing this as `token` at the top level of the handler context. In the default function definition we don't type the `bot_access_token` even though it comes in through `body.event`. Is there a reason we want to here?

Reply:
Personally, I don't think we should surface this from the SDK. This follows the pattern we have for the primary function, and it might be confusing for devs to see two different tokens